tests/test_fundamental_params.py

```diff
@@ -3,48 +3,87 @@
+def sub_spec():
+    # Initialize the substellar SED
+    sub_spec = sed.SED(name="2MASS J04151954-0935066", substellar=True)
+    # Load in a spectrum to test
+    sub_spec.add_spectrum_file(
+        "tests/data/2MASS_J04151954-0935066_apparent_SED.txt",
+        wave_units=u.micron,
+        flux_units=u.erg / u.s / u.cm ** 2 / u.AA,
+    )
+    return sub_spec
+
+
 @pytest.fixture
 def spec():
+    # Initialize the non_substellar SED
+    spec = sed.SED(name="2MASS J04151954-0935066", substellar=False)
     # Load in a spectrum to test
-    spec = sed.SED(name = "2MASS J04151954-0935066")
     spec.add_spectrum_file(
         "tests/data/2MASS_J04151954-0935066_apparent_SED.txt",
         wave_units=u.micron,
-        flux_units=u.erg / u.s / u.cm**2 / u.AA,
+        flux_units=u.erg / u.s / u.cm ** 2 / u.AA,
     )
-
-    # Calculate the fundamental parameters using default settings
-    spec.results
     return spec
 
 
-def test_just_spectrum(spec):
-    assert np.isclose(spec.fbol[0], 1.9184645e-12 * u.erg / u.s / u.cm**2)
-    assert np.isclose(spec.fbol[1], 5.26164965e-15 * u.erg / u.s / u.cm**2)
-    assert spec.mbol[0] == 17.811
-    assert spec.mbol[1] == 0.003
-    assert spec.Lbol is None
-    assert spec.radius is None
-    assert spec.Teff is None
-    assert spec.logg is None
-
-
-def test_age_distance(spec):
-    spec.age = 4.5 * u.Gyr, 0.1 * u.Gyr
-    spec.distance = 10 * u.pc, 0.1 * u.pc
-    spec.results
-
-    assert np.isclose(spec.fbol[0], 1.9184645e-12 * u.erg / u.s / u.cm**2)
-    assert np.isclose(spec.fbol[1], 5.26164965e-15 * u.erg / u.s / u.cm**2)
-    assert spec.mbol[0] == 17.811 
-    assert spec.mbol[1] == 0.003
-    assert np.isclose(spec.Lbol[0], 2.29543367e+28 * u.erg / u.s)
-    assert np.isclose(spec.Lbol[1], 4.63561547e+26  * u.erg / u.s, rtol=0.05)
-    assert spec.Teff == (903 * u.K,  4.0 * u.K, 4.0 * u.K)
-    assert spec.radius == (0.1 * u.Rsun, 0.0 * u.Rsun, 0.0 * u.Rsun)
-    assert spec.logg is None
-
-
-def test_radius(spec):
-     
-    spec.infer_radius(infer_from='evo_model')
+@pytest.mark.parametrize('seds', ['sub_spec', 'spec'])
+def test_just_spectrum(seds, request):
+    sed = request.getfixturevalue(seds)
+    sed.results
+    assert np.isclose(sed.fbol[0], 1.9184645e-12 * u.erg / u.s / u.cm ** 2)
+    assert np.isclose(sed.fbol[1], 5.26164965e-15 * u.erg / u.s / u.cm ** 2)
+    assert sed.mbol == (17.811, 0.003)
+    assert np.isclose(sed.Lbol[0], 7.56290304e+27 * u.erg / u.s, rtol=0.05)
+    assert np.isclose(sed.Lbol[1], 2.38246507e+26 * u.erg / u.s, rtol=0.05)
+    if seds == 'sub_spec':
+        assert sed.radius == (0.973 * u.Rjup, 0 * u.Rjup, 0 * u.Rjup)
+        assert np.isclose(sed.Teff[0], 684 * u.K, rtol=0.5)
+        assert sed.Teff[1] == (5 * u.K, 5 * u.K)
+    if seds == 'spec':
+        assert sed.radius == (0.1 * u.solRad, 0 * u.solRad, 0 * u.solRad)
+        assert np.isclose(sed.Teff[0], 684 * u.K, rtol=0.5)
+        assert sed.Teff[1] == (5 * u.K, 5 * u.K)
```

Comment:
The idea is that in this test, `just_spectrum`, radius, Teff, etc will all be zero because no age or distance has yet been provided. radius and Teff should be tested in the next test called "add_distance".

Reply:
yes, I wanted to ask you about that because even though we don't provide one, sedkit automatically finds one from simbad and adds it, so I think for now I am going to manually set the distance to None. Even though an age is not being provided, it still manages to calculate these values based on the calculated Lbol_sun and it can also use the spt relations.